cli/src/main/java/com/box/l10n/mojito/cli/command/PullCommand.java

```diff
@@ -282,16 +439,38 @@ void generateLocalizedFilesWithLocaleMaping(
       Repository repository, FileMatch sourceFileMatch, List<String> filterOptions)
       throws CommandException {
 
-    logger.debug("Generate localzied files with locale mapping");
+    logger.debug("Generate localized files with locale mapping");
 
-    for (Map.Entry<String, String> localeMapping : localeMappings.entrySet()) {
-      String outputBcp47tag = localeMapping.getKey();
-      RepositoryLocale repositoryLocale = getRepositoryLocaleForOutputBcp47Tag(outputBcp47tag);
-      generateLocalizedFile(
-          repository, sourceFileMatch, filterOptions, outputBcp47tag, repositoryLocale);
+    List<RepositoryLocale> repositoryLocales =
+        localeMappings.entrySet().stream()
+            .map(entry -> getRepositoryLocaleForOutputBcp47Tag(entry.getKey()))
+            .distinct()
+            .collect(Collectors.toList());
+
+    if (isParallel) {
+      generateLocalizedFiles(repository, sourceFileMatch, filterOptions, repositoryLocales);
```

Comment:
should this mention "parallel" in the method name?

Reply:
I've moved this method to a `PullCommandParallel` class now so I don't think the rename is still required